cmd/notice-file-generator/main.go

```diff
@@ -68,11 +69,23 @@ func main() {
 	if err = CreateNoticeDir(config); err != nil {
 		log.Fatalf("Error occured while creating work folder %s:%v", config.Name, err)
 	}
+
+	var wg sync.WaitGroup
+
 	for _, d := range dependencies {
-		if err = d.Generate(config); err != nil {
-			log.Printf("Error occured while generating notice.txt %s:%v", d.Name, err)
-		}
+		wg.Add(1)
+		d := d
+		go func() {
```

Comment:
Ideally you want to handle the errors which they are going to arise. It will be a good idea to use the [errgroup](https://bostonc.dev/blog/go-errgroup)